src/llmq/quorums.cpp

```diff
@@ -502,14 +500,50 @@ std::vector<CQuorumCPtr> CQuorumManager::ScanQuorums(Consensus::LLMQType llmqTyp
         return {};
     }
 
-    const CBlockIndex* pIndexScanCommitments{pindexStart};
+    const CBlockIndex* pindexStore{pindexStart};
+    const auto& llmq_params_opt = GetLLMQParams(llmqType);
+    assert(llmq_params_opt.has_value());
+
+    {
+        // Quorum sets can only change during the mining phase of DKG.
+        // Find the closest known block index.
+        const int quorumCycleStartHeight = pindexStart->nHeight - (pindexStart->nHeight % llmq_params_opt->dkgInterval);
+        const int quorumCycleMiningStartHeight = quorumCycleStartHeight + llmq_params_opt->dkgMiningWindowStart;
+        const int quorumCycleMiningEndHeight = quorumCycleStartHeight + llmq_params_opt->dkgMiningWindowEnd;
+
+        if (pindexStart->nHeight < quorumCycleMiningStartHeight) {
+            // too early for this cycle, use the previous one
+            pindexStore = pindexStart->GetAncestor(quorumCycleMiningEndHeight - llmq_params_opt->dkgInterval);
+        } else if (pindexStart->nHeight > quorumCycleMiningEndHeight) {
+            // we are past the mining phase of this cycle, use it
+            pindexStore = pindexStart->GetAncestor(quorumCycleMiningEndHeight);
+        }
+        // everything else is inside the mining phase of this cycle, no pindexStore adjustment needed
+
+        if (pindexStore == nullptr || !utils::IsQuorumTypeEnabled(llmqType, *this, pindexStore)) {
+            return {};
+        }
+    }
+
+    const CBlockIndex* pIndexScanCommitments{pindexStore};
```

Comment:
not sure if this is valuable or not, just an idea
```suggestion
    gsl::not_null<const CBlockIndex*> pIndexScanCommitments{pindexStore};
```